tests/base/test_pytorch_transforms.py

```diff
@@ -40,3 +138,387 @@ def forward(self, x):
     assert transformed
     y2 = model(x)
     assert torch.all(y2 == x)
+
+
+@pytest.mark.parametrize("input_size", [random.randint(0, 10)], ids=lambda x: "input_size=" + str(x))
+@pytest.mark.parametrize("hidden_size", random.sample([4, 64, 2048, 4096], 2), ids=lambda x: "hidden_size=" + str(x))
+@pytest.mark.parametrize("module", CustomOpsTransform._module_mapping.keys(), ids=lambda x: "module=" + x.__name__)
+def test_custom_ops_transform(module: nn.Module, hidden_size: int, input_size: int) -> None:
+    """Test custom Ops transform individually
+
+    Args:
+        module (nn.Module): Pytorch module
+        hidden_size (int): hidden_size for RMSNorm operation
+        input_size (int): Random inputs shape for testing
+    """
+    model = module(hidden_size=hidden_size)
+    rand_data = torch.rand(input_size, hidden_size)
+
+    original_output = model(rand_data)
+
+    model, transformed = CustomOpsTransform.apply(model)
+    assert transformed
+
+    transformed_model_output = model(rand_data)
+
+    assert not isinstance(model, module)
+    assert torch.all(original_output == transformed_model_output)
+
+
+@pytest.mark.parametrize("input_len", [8], ids=lambda x: "input_len=" + str(x))
+@pytest.mark.parametrize("hidden_size", [128], ids=lambda x: "hidden_size=" + str(x))
+@pytest.mark.parametrize("intermediate_size", [512], ids=lambda x: "intermediate_size=" + str(x))
+@pytest.mark.parametrize("num_key_value_heads", [8, 32], ids=lambda x: "num_key_value_heads=" + str(x))
+@pytest.mark.parametrize("num_attention_heads", [32], ids=lambda x: "num_attention_heads=" + str(x))
+@pytest.mark.parametrize("ctx_len", [32], ids=lambda x: "ctx_len=" + str(x))
+@pytest.mark.parametrize("num_hidden_layers", [1, 3], ids=lambda x: "num_hidden_layers=" + str(x))
+def test_kv_cache_transform_llama(
```

Comment:
There should be a better way to do this than copy paste this function and just change the names. You should be able to use `transformers.AutoConfig` along with `transformers.AutoModelForCausalLM`.

Reply:
We are not directly copying the code.
The names of input arguments are different for different models.
and we are changing the arguments to be of smaller size so that the runtime reduces.
If we had to combine all these tests into one tests, we still will have to write these parameters names and values somewhere.
It's easier that pytorch handles this than us doing it.